src/main/java/edu/umich/its/spe/SPEMaster.java

```diff
@@ -0,0 +1,378 @@
+	 * timestamp will be returned. An empty result is reasonable.
+	 * The format of the time stamp is: 2017-04-01 18:00:00.  
+	 * TODO: timestamp default value?
+	 */ 
+
+	public String getSPEGrades(String gradedAfterTime) throws SPEEsbException {
+
+		HashMap<String, String> values = setupGetGradesCall(gradedAfterTime);
+
+		WAPIResultWrapper grades = speesb.getGradesViaESB(values);
+		// check for possibility of no new grades.
+		if (grades.getStatus() == 404) {
+			return "[]";
+		}
+
+		return grades.getResult().toString();
+	}
+
+	//{"Meta":{"Message":"COMPLETED","httpStatus":200},"Result":{"putPlcExamScoreResponse":{"putPlcExamScoreResponse":{"Status":"SUCCESS","Form":7,"ID":"abc"},"@schemaLocation":"http://mais.he.umich.edu/schemas/putPlcExamScoreResponse.v1 http://csqa9ib.dsc.umich.edu/PSIGW/PeopleSoftServiceListeningConnector/putPlcExamScoreResponse.v1.xsd"}}}
+
+	//{"putPlcExamScoreResponse":{"putPlcExamScoreResponse":{"Status":"SUCCESS","Form":7,"ID":"abc"
+
+	/*********** format assignments from the data warehouse ********/
+	
+	// After getting the JSON from the data warehouse format the SPE grades.
+	protected ArrayList<HashMap<String,String>> convertSPEGradesFromDataWarehouseJSON(String assignmentJSON) {
+	
+		M_log.debug("cSGFDWJ: assignmentJSON: {}",assignmentJSON);
+		// if nothing in it then return an empty list.
+		if (assignmentJSON == null || assignmentJSON.length() <=2) {
+			ArrayList<HashMap<String,String>> emptyArrayList = new ArrayList<HashMap<String,String>>();
+			return emptyArrayList;
+		}
+	
+		JSONArray assignments = parseCanvasAssignmentJSON(assignmentJSON);
+		ArrayList<HashMap<String,String>> gradeMapList = convertAssignmentsToGradeMaps(assignments);
+		M_log.debug("cSGFDWJ: gradeMaplist: {}",gradeMapList);
+		return gradeMapList;
+	}
+
+	// convert the list of DataWarehouse assignments to a list of grade maps for MPathways.
+	protected ArrayList<HashMap<String, String>> convertAssignmentsToGradeMaps(JSONArray canvasAssignments) {
+		ArrayList<HashMap<String,String>> gradeMapList = new ArrayList<HashMap<String, String>>();
+	
+		// JSONArray is not iterable so can't use the nice "for each" syntax.
+		for(int i = 0; i<canvasAssignments.length(); i++) {
+			JSONObject assignment = canvasAssignments.getJSONObject(i);
+			HashMap<String, String> grademap = convertAssignmentToGradeMap(assignment);
+			gradeMapList.add(grademap);
+		}
+	
+		return gradeMapList;
+	}
+
+	// convert a single JSON version of an assignment to a grademap.
+	static protected HashMap<String, String> convertAssignmentToGradeMap(JSONObject assignment) throws JSONException {
+		HashMap<String,String> grademap = new HashMap<String,String>();
+		// Score may be read as a number instead of a string so pull out as an object and convert to a string.
+		grademap.put("Score",JSONObject.valueToString(assignment.get("Score")));
+		grademap.put("Unique_Name", assignment.getString("Unique_Name"));
```

Comment:
`Score` and `Unique_Name` can also be made constants as these are reference again in `setupPutGradeCall` method.
I see some of the Hard coded string and used all over place, So constant can be made for all those to avoid repeating then. Better if we have a Util class that handles all these, that way we have one class to go for